### PR TITLE
Store Email Settings: Improve configuration

### LINKED ISSUE #5623: Introduce a setting for Server Setting SMPT to be used for all stores in Server Settings > Email
### Discussed in https://github.com/btcpayserver/btcpayserver/discussions/5621

<div type='discussions-op-text'>

<sup>Originally posted by **iBobik** December 26, 2023</sup>
Currently each store requires to configure own SMTP settings, but it is very impractical if I am ok with using same settings for more stores - password cannot be recovered from existing store to copy it into new store and many SMTP servers cannot recover password too (just generate new password).

Could it be optional to configure SMTP? When not configured, it will use server global SMTP.</div>

### PR DESCRIPTION
**Pull request by dennisreimann**
This works with the existing settings and provides better guidance about the different store email cases. Closes #5623.

## No Fallback

- Either the SMTP settyings on the server level are not configured or the "Disable stores from using the server's email settings as backup" policy is active. 
- Requires configuration of SMTP settings on the store level: No option for using the server fallback.

no-fallback

## Fallback is present and used

- Toggle indicates the availability of providing custom SMTP settings for the store
- By default the server's SMTP settings will be used

uses-fallback

## Fallback is present but custom settings are provided

with-fallback-but-custom

### CODE REVIEW

File: BTCPayServer.Tests/SeleniumTests.cs [hunk 1]
```diff
@@ -3110,15 +3123,13 @@ private static void CanBrowseContent(SeleniumTester s)
 
         private static void CanSetupEmailCore(SeleniumTester s)
         {
+            s.Driver.ScrollTo(By.Id("Settings_Server"));
```
Comment (dennisreimann) [inline BTCPayServer.Tests/SeleniumTests.cs:3126]:
```suggestion
            s.Driver.ScrollTo(By.Id("QuickFillDropdownToggle"));
```

**Review by dstrukt — APPROVED** [review]
This is so much better!

I didn't run it through a rigorous technical review, but functionally it worked as expected.

Will be a nice quality of life improvement for admins - tACK!

**Review by Kukks — APPROVED** [review]

**Comment by dennisreimann** [comment]
Feedback from design call: @pavlenex suggested to move (or mirror) the policy setting on the server SMTP setup view.

I think we should also invert the wording to make it "enable" instead of "disable" on the server level, as this makes it easier to think about it.

**Comment by dennisreimann** [comment]
## Server policies 

n1

## Store email settings

n2

**Comment by dstrukt** [comment]
Few nits:

- [ ] Shouldn't this be under `Email Settings` instead of `Notification Settings` in the Server Policies?

Copy suggestion, but don't necessarily know if it's any better, but is more consistent with the custom messages..`Use the Server's SMTP settings across all stores to send emails. This can be customized at the store level.` cc @pavlenex 

Otherwise, tACK!

**Review by dstrukt — CHANGES_REQUESTED** [review]
Just flagging the above, this could be shipped as is!

**Comment by dennisreimann** [comment]
As discussed yesterday. This is now ready to be merged. 

grafik

**Pull request merged**